internal/services/network/vpn_gateway_nat_rule_resource.go

```diff
@@ -53,21 +54,63 @@ func resourceVPNGatewayNatRule() *pluginsdk.Resource {
 				ValidateFunc: validate.VpnGatewayID,
 			},
 
-			"external_address_space_mappings": {
-				Type:     pluginsdk.TypeSet,
-				Required: true,
-				Elem: &pluginsdk.Schema{
-					Type:         pluginsdk.TypeString,
-					ValidateFunc: validation.IsCIDR,
+			"external_mapping": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Computed: !features.FourPointOhBeta(),
+				AtLeastOneOf: func() []string {
```

Comment:
All of `AtLeastOneOf` should be swapped to `ExactlyOneOf` since it's required and we don't want someone to put both `external_address_space_mappings` and `external_mapping` in

Reply:
Updated